### PR TITLE
Some improvement for the Docker image.

### PR DESCRIPTION
**Pull request by blasvicco**
### Change log
- 2018-09-26 Using the small Docker image Alpine instead of Debian.
- 2018-09-26 Using docker-compose to build/up/down the image.
- 2018-09-26 Grouping project files inside the app folder.

### Motivation
Using a small docker image.
Using docker-compose to sync the app folder and its content between the docker image and the host machine.
Avoiding clone the project inside the docker image.
Separating the project file from the docker files.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,82 +35,82 @@ schema validation.
...
 2. Clone or download *bfx-api-cpp* repository.
-3. Add `key-secret` file in `bfx-api-cpp/doc` directory. (or edit `example.cpp` so that it doesn't use `key-secret` file)
-4. Peek into self-documented `<your_project_dir>/src/example.cpp`.
-5. Create `build` directory
+3. Add `key-secret` file in `bfx-api-cpp/app/doc` directory. (or edit `example.cpp` so that it doesn't use `key-secret` file)
+4. Peek into self-documented `<your_project_dir>app/src/example.cpp`.
+5. Build `example` binary
 
- `$ mkdir <your_project_dir>/build`
+ `$ cd <your_project_dir>app/build && cmake .. && make`
```
Comment (MMquant) [inline README.md:45]: List item 6 is missing.

File: README.md [hunk 2]
```diff
@@ -35,82 +35,82 @@ schema validation.
...
-6. Build `example` binary
-
- `$ cd <your_project_dir>/build && cmake .. && make`
- 
-7. Run `example` binary
+7. Run `example` binary from `<your_project_dir>app/bin`
 
  `$ ./example`
 
 ### How to Build'n'Run `src/example.cpp` in Docker container
```
Comment (MMquant) [inline README.md:52]: `src/example.cpp` -> `app/src/example.cpp` ?

File: README.md [hunk 3]
```diff
@@ -35,82 +35,82 @@ schema validation.
...
-	```
-7. Run `example` binary
+ ```BASH
+ cd /home/bfx-cpp-api/app/build
+ cmake ..
+ make
+ ```
 
-  `$ ./example`
+6. Run `example` binary
```
Comment (MMquant) [inline README.md]: List item should be 7.

File: README.md [hunk 4]
```diff
@@ -35,82 +35,82 @@ schema validation.
...
 
+	```BASH
```
Comment (MMquant) [inline README.md]: Fix indentation

File: README.md [hunk 5]
```diff
@@ -35,82 +35,82 @@ schema validation.
...
+		}
+		else
+		{
+		    // Inspect errors
+		    cout << bfxAPI.getBfxApiStatusCode() << endl;
+		    cout << bfxAPI.getCurlStatusCode() << endl;
+		}
+	```
 
 See self-explanatory `src/example.cpp` for general usage and more requests.
```
Comment (MMquant) [inline README.md:123]: `src/example.cpp` -> `app/src/example.cpp`

**Comment by MMquant** [comment]
Thanks for pull requests. However I would be very happy if you make pull requests to Develop branch which has currently same HEAD as Master branch. My develop policy is no direct pull requests into Master.

**Comment by blasvicco** [comment]
Sure thing!
I updated the PR to develop and fixed the reviews.

**Pull request merged**